### PR TITLE
feat(proto): extends Rustacean struct

### PR DESCRIPTION
**Pull request by luchosr**

### CODE REVIEW

File: crates/proto/src/lib.rs [hunk 1]
```diff
@@ -4,5 +4,12 @@ use serde::{Deserialize, Serialize};
...
     pub name: String,
     pub image: String,
+    pub bio:String,
     pub github_url: String,
+    pub email:String,
+    pub geographical_location: String,
+    pub personal_website:Option<String>,
+    pub mastodon_fediverse:Option<String>, 
+    pub IRC:Option<String>,
+    pub preferred_social_network:Option<String>,
```
Comment (EstebanBorai) [inline crates/proto/src/lib.rs:14]: I think Social Networks could have an structure where we can determinate:

- Provider
- Username

For the ones where username can be easily build by appending/prefixing it to a URL.

For instance on sites like X and Instagram one can use:

https://x.com/EstebanBorai

Theres other social networks where a channel is specified, for instance in the Fediverse you can have any server in the network so you find solutions like Mastodon/Bluesky where the URL may not be deterministic, for instance Hachyderm.io.

These cases could be a "Custom" for now.

The structure could be:

```rust
enum SocialNetwork {
  Instagram {
    username: String,
  }
  X {
    username: String
  }
  Custom {
    url: Url
  }
}
```

Then instead of `preferred_social_network` we could have `social_networks` and use a `HashSet` instead to make sure they are not duplicated on storage.

For the URL we can use this struct: https://docs.rs/url/latest/url/struct.Url.html

File: crates/proto/src/lib.rs [hunk 2]
```diff
@@ -4,5 +4,12 @@ use serde::{Deserialize, Serialize};
 pub struct Rustacean {
     pub name: String,
     pub image: String,
+    pub bio:String,
     pub github_url: String,
+    pub email:String,
+    pub geographical_location: String,
+    pub personal_website:Option<String>,
+    pub mastodon_fediverse:Option<String>, 
+    pub IRC:Option<String>,
```
Comment (EstebanBorai) [inline crates/proto/src/lib.rs:13]: The IRC chat should have two properties:

- The user's IRC ID
- User's channels

I think we could have the struct:

```struct
struct Irc {
  id: String,
  channels: HashSet<String>,
}
```

Comment (EstebanBorai) [inline crates/proto/src/lib.rs:13]:
```suggestion
    pub irc: Option<Irc>,
```

File: crates/proto/src/lib.rs [hunk 3]
```diff
@@ -4,5 +4,12 @@ use serde::{Deserialize, Serialize};
 pub struct Rustacean {
     pub name: String,
     pub image: String,
+    pub bio:String,
     pub github_url: String,
+    pub email:String,
+    pub geographical_location: String,
+    pub personal_website:Option<String>,
+    pub mastodon_fediverse:Option<String>,
```
Comment (EstebanBorai) [inline crates/proto/src/lib.rs:12]: I think this could go in `SocialNetwork` hashset

File: crates/proto/src/lib.rs [hunk 4]
```diff
@@ -4,5 +4,12 @@ use serde::{Deserialize, Serialize};
 pub struct Rustacean {
     pub name: String,
     pub image: String,
+    pub bio:String,
     pub github_url: String,
+    pub email:String,
+    pub geographical_location: String,
+    pub personal_website:Option<String>,
```
Comment (EstebanBorai) [inline crates/proto/src/lib.rs:11]:
```suggestion
    pub personal_website: Option<Url>,
```

File: crates/proto/src/lib.rs [hunk 5]
```diff
@@ -4,5 +4,12 @@ use serde::{Deserialize, Serialize};
 pub struct Rustacean {
     pub name: String,
     pub image: String,
+    pub bio:String,
     pub github_url: String,
+    pub email:String,
+    pub geographical_location: String,
```
Comment (EstebanBorai) [inline crates/proto/src/lib.rs:10]: For this I think it would be nice to have normalized values.

Listing a full set of locations could be memory expensive, using an API as well, so for now I think we could have an approximate location and limit to County and Continent.

These could be filled as pairs, Continent could be "required" but Country optional.

```suggestion
    pub location: Location,
```

and then

```rust
struct Location {
  continent: Continent,
  country: Country,
}
```

The downside is that we would have to write these by hand (or generate these using a from a CSV), but the good part is that we would do it once (and perhaps when some of this data changes due to political reasons).

The benefit here is that normalized data will allow us to perform better filtering later, as well as taking advantage of its precision for things like adding emojis for instance or locating in a map, or perhaps reusing some ISO bits to perform timezone matching for a range of TZs in a Country space.

File: crates/proto/src/lib.rs [hunk 6]
```diff
@@ -4,5 +4,12 @@ use serde::{Deserialize, Serialize};
 pub struct Rustacean {
     pub name: String,
     pub image: String,
+    pub bio:String,
     pub github_url: String,
```
Comment (EstebanBorai) [inline crates/proto/src/lib.rs:8]: Not part of this PR but given that we rely on GitHub for identity validation, I would just leave this either as `id` or `gh_user` (as it is a bit more compact)

File: crates/proto/src/lib.rs [hunk 7]
```diff
@@ -4,5 +4,12 @@ use serde::{Deserialize, Serialize};
 pub struct Rustacean {
     pub name: String,
     pub image: String,
```
Comment (EstebanBorai) [inline crates/proto/src/lib.rs:6]: As well, not part of this PR, but I think if a user dont want to show off their picture, they could leave this as `None`, wdyt?

**Review by EstebanBorai — COMMENTED** [review]
Notes on fields and how to optimize data structure for writes.

Given that we are using plaintext for storage, I suggest we enforce a schema on write, allowing us to rely on this data on reads.

Another advantage is indexing, it should be possible to filter by certain fields like location for insance, for these cases the best we can do is normalize data so we have more precision.

For social media we can easily avoid duplicates using a `HashSet`.

File: crates/proto/src/lib.rs (hunk 7 above)
Comment (luchosr) [reply crates/proto/src/lib.rs:6]: SGTM.
